zenodo_uploader.py

```diff
@@ -1,81 +1,278 @@
+            depositions.append(deposition)
+
+    if not depositions:
+        print(f"No records found for search: '{title}'")
+        return None, None, None
+
+    deposition = sorted(
+        depositions,
+        key=lambda deposition: deposition["metadata"]["publication_date"],
+        reverse=True,
+    )[0]
+
+    print(f"Best match is deposition: {deposition['id']}")
+    return (
+        deposition["id"],
+        deposition["links"]["bucket"],
+        deposition["links"]["html"].replace("deposit", "record"),
+    )
+
+def create_new_version(
+    deposition_id, token, zenodo_server="https://sandbox.zenodo.org/api/"
+):
+    url = f"{zenodo_server}deposit/depositions/{deposition_id}/actions/newversion"
     r = requests.post(
-        zenodo_server,
-        params=params,
+        url,
+        params={"access_token": token},
+    )
+    r.raise_for_status()
+
+    deposition = r.json()
+    new_deposition_url = deposition["links"]["latest_draft"]
+    new_deposition_id = new_deposition_url.split("/")[-1]
+
+    r = requests.get(
+        f"{zenodo_server}deposit/depositions/{new_deposition_id}",
+           params={"access_token": token},
+    )
+    r.raise_for_status()
+    deposition = r.json()
+
+    return (
+        deposition["id"],
+        deposition["links"]["bucket"],
+        deposition["links"]["html"].replace("deposit", "record"),
+    )
+
+def create_new_deposition(token, zenodo_server="https://sandbox.zenodo.org/api/"):
+    url = f"{zenodo_server}deposit/depositions"
+    r = requests.post(
+        url,
+        params={"access_token": token},
         json={},
-        headers=headers,
+        headers={"Content-Type": "application/json"},
     )
-    if r.status_code != 201:
-        raise RuntimeError(f"The status code for the request is {r.status_code}.\nMessage: {r.text}")
+    r.raise_for_status()
 
-    return_json = r.json()
-    deposition_id = return_json["id"]
-    bucket_url = return_json["links"]["bucket"]
+    deposition = r.json()
 
-    filebase = os.path.basename(file_name_to_upload)
+    return (
+        deposition["id"],
+        deposition["links"]["bucket"],
+        deposition["links"]["html"].replace("deposit", "record"),
+    )
 
-    file_url = return_json["links"]["html"].replace('deposit', 'record')
+def upload_to_zenodo(
+    deposition_id,
+    filename,
+    bucket_url,
+    file_url,
+    filebase,
+    env_name,
+    token,
+    zenodo_server="https://sandbox.zenodo.org/api/",
+):
+    if filename.endswith(".tar.gz"):
+        response = requests.get(
+            f"{zenodo_server}deposit/depositions/{deposition_id}",
+            params={"access_token": token},
+            headers={"Content-Type": "application/json"},
+        )
+        response.raise_for_status()
+
+        print(f"Comparing {filename} checksum with files on zenodo...")
+        files_checksums = [file["checksum"] for file in response.json()["files"]]
+        with open(f"{env_name}-md5sum.txt", "r") as fp:
```

Comment:
This can be a quick workaround with the assumption the checksum file resides in the same location as the .tar.gz one.
```suggestion
        with open(f"{os.path.dirname(filename)}/{env_name}-md5sum.txt", "r") as fp:
```